integration-scripts/src/countTransactionEvents.ts

```diff
@@ -0,0 +1,61 @@
+import { ethers } from 'ethers'
+import chalk from 'chalk'
+import { registerPromiseHandler, createProvider, getArgs } from './common'
+
+async function main() {
+  registerPromiseHandler()
+  // see: runlog_test for these variable names
+  const args = getArgs(['RUN_LOG_ADDRESS', 'txid'])
+
+  await countTransactionEvents({
+    fromAddress: args.RUN_LOG_ADDRESS,
+    txId: args.txid,
+  })
+}
+main()
+
+interface Args {
+  txId: string
+  fromAddress: string
+}
+async function countTransactionEvents({ fromAddress, txId }: Args) {
+  const provider = createProvider()
+  const receipt = await provider.getTransactionReceipt(txId).catch(e => {
+    console.error('Error getting transaction receipt')
+    console.error(chalk.red(e))
+    throw e
+  })
+  const count = countLogsEmittedBy(fromAddress, receipt.logs)
+
+  emitFulfillmentEvent({ count, fromAddress, txId })
+}
+
+/**
+ * Count the number of logs emitted by a particular address
+ * @param address The address to match against log emissions
+ * @param logs The logs emitted by a transaction
+ */
+function countLogsEmittedBy(
+  address: string,
+  logs?: ethers.providers.Log[],
+): number {
+  if (!logs) {
+    return 0
+  }
+
+  return logs.reduce((n, l) => (l.address === address ? n + 1 : n), 0)
+}
+
+interface FulfillmentEvent {
+  txId: string
+  fromAddress: string
+  count: number
+}
+
+/**
+ * Log a fulfillment event in a suitable format for our integration tests to capture via `grep`
```

Comment:
I extracted this log out because it has special meaning, it needs to stay the exact same for our integration scripts to work. Let me know if you still think I should change it though, would be happy to.

Reply:
Makes sense. Maybe just add a remark to that effect to its docstring.

Reply:
Hm, I did write "in a suitable format for our integration tests to capture via `grep`", do you think it should be worded differently?